ui/ducks/app/app.ts

```diff
@@ -1,12 +1,85 @@
+import { AnyAction, Action } from 'redux';
+import { PayloadAction } from '@reduxjs/toolkit';
 import {
   WebHIDConnectedStatuses,
   HardwareTransportStates,
 } from '../../../shared/constants/hardware-wallets';
+import { RPCDefinition } from '../../../shared/constants/network';
 import * as actionConstants from '../../store/actionConstants';
 
-export default function reduceApp(state = {}, action) {
+interface AppState {
+  shouldClose: boolean;
+  menuOpen: boolean;
+  modal: {
+    open: boolean;
+    modalState: {
+      name: string | null;
+      props: Record<string, any>;
+    };
+    previousModalState: {
+      name: string | null;
+    };
+  };
+  alertOpen: boolean;
+  alertMessage: string | null;
+  qrCodeData: {
+    type?: string | null;
+    values?: { address?: string | null };
+  } | null;
+  networkDropdownOpen: boolean;
+  accountDetail: {
+    subview?: string;
+    accountExport?: string;
+    privateKey?: string;
+  };
+  isLoading: boolean;
+  loadingMessage: string | null;
+  scrollToBottom: boolean;
+  warning: string | null | undefined;
+  buyView: Record<string, any>;
+  isMouseUser: boolean;
+  defaultHdPaths: {
+    trezor: string;
+    ledger: string;
+    lattice: string;
+  };
+  networksTabSelectedRpcUrl: string | null;
+  loadingMethodData: boolean;
+  requestAccountTabs: Record<string, number>; // [url.origin]: tab.id
+  openMetaMaskTabs: Record<string, boolean>; // openMetamaskTabsIDs[tab.id]): true/false
+  currentWindowTab: Record<string, any>; // tabs.tab https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/tabs/Tab
+  showWhatsNewPopup: boolean;
+  singleExceptions: {
+    testKey: string | null;
+  };
+  gasLoadingAnimationIsShowing: boolean;
+  smartTransactionsError: string | null;
+  smartTransactionsErrorMessageDismissed: boolean;
+  ledgerWebHidConnectedStatus: WebHIDConnectedStatuses;
+  ledgerTransportStatus: HardwareTransportStates;
+  newNetworkAdded: string;
+  newCollectibleAddedMessage: string;
+  removeCollectibleMessage: string;
+  portfolioTooltipWasShownInThisSession: boolean;
+  sendInputCurrencySwitched: boolean;
+  newTokensImported: string;
+  newCustomNetworkAdded: RPCDefinition | Record<string, any>;
+  onboardedInThisUISession: boolean;
+  customTokenAmount: string;
+  forgottenPassword: boolean | null;
+  txId: number | null;
+}
+
+interface AppSliceState {
+  appState: AppState;
+}
+
+export default function reduceApp(
+  state: AppSliceState,
```

Comment:
I believe state is AppState, not AppSliceState. see in here we spread ...state into appState, and we return it, we only have appState as the key due to ducks/index.js which puts it on the appState key.

Reply:
yes you are correct, modified this one and also changed default state as 
```
  const appState: AppState = {
    ...state,
    shouldClose: false,
```

Reply:
thanks for the suggestion, spread into 2 now 🙏